File: codigos_de_javascript/arreglo2.js
let arr = [5, 3, 8, 1, 2];
let tamanoOriginal = arr.length;
let elementoIns = 4;
let posicion = 2;

console.log("Arreglo original:", arr.join(" "));

let nuevoArr = [];
for (let i = 0; i < posicion; i++) {
    nuevoArr[i] = arr[i];
}
nuevoArr[posicion] = elementoIns;
for (let i = posicion; i < tamanoOriginal; i++) {
    nuevoArr[i + 1] = arr[i];
}

console.log("Arreglo con elemento insertado:", nuevoArr.join(" "));

let posEliminar = 3;
let arrEliminado = [];
for (let i = 0; i < nuevoArr.length; i++) {
    if (i !== posEliminar) {
        arrEliminado.push(nuevoArr[i]);
    }
}
console.log("Eliminacion por posicion:", arrEliminado.join(" "));

let elementoBuscar = 3;
let encontrado = false;
for (let i = 0; i < nuevoArr.length; i++) {
    if (nuevoArr[i] === elementoBuscar) {
        console.log(`Elemento ${elementoBuscar} encontrado en la posicion ${i}`);
        encontrado = true;
        break;
    }
}
if (!encontrado) {
    console.log(`Elemento ${elementoBuscar} no encontrado`);
}

File: codigos_de_python/Arreglo2.py
arr = [5, 3, 8, 1, 2]
tamanoOriginal = len(arr)
elementoIns = 4
posicion = 2

print("Arreglo original:", " ".join(map(str, arr)))

# Insertar elemento
nuevoArr = arr[:posicion] + [elementoIns] + arr[posicion:]
print("Arreglo con elemento insertado:", " ".join(map(str, nuevoArr)))

# Eliminar por posición
posEliminar = 3
arrEliminado = [nuevoArr[i] for i in range(len(nuevoArr)) if i != posEliminar]
print("Eliminacion por posicion:", " ".join(map(str, arrEliminado)))

# Búsqueda lineal
elementoBuscar = 5
encontrado = False
for i in range(len(nuevoArr)):
    if nuevoArr[i] == elementoBuscar:
        print(f"Elemento {elementoBuscar} encontrado en la posicion {i}")
        encontrado = True
        break

if not encontrado:
    print(f"Elemento {elementoBuscar} no encontrado")
